packages/@lwc/integration-karma/test/custom-elements/index.spec.js

```diff
@@ -37,54 +38,93 @@ const undefinedElementError = /(Illegal constructor|does not define a custom ele
-
-            // For LWC components, the PivotCtor is exposed externally but not the UserCtor
-            it(method, () => {
-                expect(customElements.get(tag)).toBeUndefined();
-                const promise = customElements.whenDefined(tag);
-                expect(customElements.get(tag)).toBeUndefined();
-                const elm = create(tag, Component);
-                const Ctor = customElements.get(tag);
-                expect(typeof Ctor).toEqual('function');
+        it('using CustomElementConstructor', () => {
+            // Nonce elements should be defined only once in the entire Karma test suite
+            const tagName = 'x-nonce1';
+            expect(customElements.get(tagName)).toBeUndefined();
+            const promise = customElements.whenDefined(tagName);
+            expect(customElements.get(tagName)).toBeUndefined();
+            customElements.define(tagName, Nonce1.CustomElementConstructor);
+            const elm = document.createElement(tagName);
+            const Ctor = customElements.get(tagName);
+            expect(Ctor).toBe(Nonce1.CustomElementConstructor);
+            document.body.appendChild(elm);
+            expect(elm.expectedTagName).toEqual(tagName);
+            return promise
+                .then((Ctor) => {
+                    expect(Ctor).toBe(Nonce1.CustomElementConstructor);
+                    return customElements.whenDefined(tagName);
+                })
+                .then((Ctor) => {
+                    expect(Ctor).toBe(Nonce1.CustomElementConstructor);
+                });
+        });
+
+        if (window.lwcRuntimeFlags.ENABLE_SCOPED_CUSTOM_ELEMENT_REGISTRY) {
+            it('using lwc.createElement', () => {
+                const tagName = 'x-nonce2';
+                expect(customElements.get(tagName)).toBeUndefined();
+                let resolvedCtor;
+                const promise = customElements.whenDefined(tagName).then((ctor) => {
+                    resolvedCtor = ctor;
+                });
+                expect(customElements.get(tagName)).toBeUndefined();
+                const elm = createElement(tagName, { is: Nonce2 });
+                const Ctor = customElements.get(tagName);
+                //lwc.createElement is "invisible" to customElements.get
+                expect(Ctor).toBeUndefined();
                 document.body.appendChild(elm);
-                expect(elm.expectedTagName).toEqual(tag);
-                return promise
-                    .then((Ctor) => {
-                        expect(typeof Ctor).toEqual('function');
-                        return customElements.whenDefined(tag);
+                expect(elm.expectedTagName).toEqual(tagName);
+                class MyCustomComponent extends HTMLElement {}
+                return Promise.resolve()
+                    .then(() => {
+                        expect(resolvedCtor).toEqual(undefined); // whenDefined does not leak pivot constructor
```

Comment:
This ^ verifies that `whenDefined()` does not leak a private element created by LWC. (At least insofar as we assume `Promise.resolve()` would have caught it)